tinygrad/shape/symbolic.py

```diff
@@ -134,6 +134,7 @@ def __new__(cls, expr:Optional[str], nmin:int, nmax:int):
 
   def __init__(self, expr:Optional[str], nmin:int, nmax:int):
     self.expr, self.min, self.max = expr, nmin, nmax
+    self.val: Optional[int] = None
```

Comment:
Should this be tracked in the Variable or in external context?

Example:
```
vi = Variable("i", 1, 10)
a = Tensor.rand(3, 4).reshape(vi, 2)
b = Tensor.rand(2, 4).reshape(vi, 2)
```

What happens there?

Reply:
the old `symbols` was trying to track these values in shapetracker level, which is also possible. But we still have the problem that two shapetracker can have different symbol:value dicts. Resolving at variable level feels the cleanest to me. Or are you suggesting a global variable like context?